### PR TITLE
Refactor:  Add error messages when user tries to login or signup

### PR DESCRIPTION
**Pull request by ekeric13**

### CODE REVIEW

File: client/src/app/models/user.js [hunk 1]
```diff
@@ -17,7 +17,14 @@ window.User = Backbone.Model.extend({
         window.localStorage.setItem("_user.name", JSON.parse(user).username);
         router.navigate('/', { trigger: true });
       },
-      error: function(err) { console.log(err.status + " " + err.statusText); }
+      error: function(err) {
+        console.log(err.status + " " + err.statusText);
+        var options = {
+          text: "Error: Incorrect username or password. Please try again."
+        };
+        this.formErrorView = new formErrorView(options);
```
Comment (d4ncer) [inline client/src/app/models/user.js]: Is this a good practice? To add a view inside a model? Is there some other way we could do this?

File: client/src/app/models/user.js [hunk 2]
```diff
@@ -29,7 +36,15 @@ window.User = Backbone.Model.extend({
...
         router.navigate('/', { trigger: true });
       },
-      error: function() { console.log("User signup error"); }
+      error: function() {
+        console.log("User signup error");
+        // _this.trigger("signupError", _this);
+        var options = {
+          text: "Error: Username or email is already taken. Please try again."
+        };
+        this.formErrorView = new formErrorView(options);
```
Comment (d4ncer) [inline client/src/app/models/user.js]: Same comment as above.

Comment (d4ncer) [inline client/src/app/models/user.js]: In addition, ideally the error message should be the same as what the server sends back.

File: client/src/app/models/user.js (hunk 1 above)
Comment (ekeric13) [inline client/src/app/models/user.js]: I was trying to trigger an event that we would listen to in a view. But that would require the to be instantiated somewhere else and with the user model. maybe put instantiating the error form in the utils?

**Pull request closed without merging**